### DISCUSSION TITLE
Can diesel perform migrations without diesel_cli

### DISCUSSION THREAD

**Discussion by alimehasin** [diesel-rs/diesel#3245] [Q&A] [answered]
Is it possible to perform migrations using diesel without the cli
A user case being perform migrations after the user installs the program for example

**Comment by brocaar** [comment, marked as answer]
Yes, you can use the `diesel_migrations` crate for embedding the migrations into your application + apply these. See:
https://docs.rs/diesel_migrations/2.0.0-rc.0/diesel_migrations/macro.embed_migrations.html
